Fix: Tab persistence service date title migration tests are time-dependent

## Problem

The tab persistence service has 3 failing tests related to date node title migration:

```
FAIL src/tests/services/tab-persistence-service.test.ts > TabPersistenceService > date node title migration > recomputes date titles on load
AssertionError: expected 'Yesterday' to be 'Today'

FAIL src/tests/services/tab-persistence-service.test.ts > TabPersistenceService > date node title migration > recomputes tomorrow date title correctly
AssertionError: expected 'Today' to be 'Tomorrow'

FAIL src/tests/services/tab-persistence-service.test.ts > TabPersistenceService > date node title migration > recomputes yesterday date title correctly
AssertionError: expected '2025-11-22' to be 'Yesterday'
```

## Root Cause

These tests are **time-dependent** - they compute relative date labels ("Today", "Tomorrow", "Yesterday") dynamically, but the test expectations are hardcoded to specific dates. When the current date crosses a day boundary, the tests fail because:

1. Tests expect a specific relative date label
2. The actual computed date has shifted due to timezone or date boundary issues
3. The test framework doesn't mock the current date/time

## Solution

The tests need to be made time-independent by:
- Mocking `Date.now()` or the current date in tests
- OR dynamically computing expected values based on actual current date
- OR using fixed dates with timezone-aware assertions

## Acceptance Criteria

- [ ] Tests use mocked/fixed dates instead of relying on system date
- [ ] All 3 tests pass consistently regardless of when they're run
- [ ] Date boundary transitions don't cause test failures

--- a/packages/desktop-app/src/tests/services/tab-persistence-service.test.ts
+++ b/packages/desktop-app/src/tests/services/tab-persistence-service.test.ts
@@ -3,6 +3,9 @@ import { TabPersistenceService } from '$lib/services/tab-persistence-service';
 import type { TabState } from '$lib/stores/navigation';
 
 describe('TabPersistenceService', () => {
+  // Fixed reference date for consistent testing: 2025-11-24 (Monday)
+  const REFERENCE_DATE = new Date('2025-11-24T12:00:00Z');
+
   beforeEach(() => {
     // Clear localStorage before each test
     localStorage.clear();
@@ -10,6 +13,8 @@ describe('TabPersistenceService', () => {
     TabPersistenceService.flush();
     // Use fake timers for debounce testing
     vi.useFakeTimers();
+    // Set system time to fixed date so relative date calculations are consistent
+    vi.setSystemTime(REFERENCE_DATE);
   });
 
   afterEach(() => {
@@ -413,7 +418,8 @@ describe('TabPersistenceService', () => {
 
   describe('date node title migration', () => {
     it('recomputes date titles on load', () => {
-      const today = new Date().toISOString().split('T')[0];
+      // Use REFERENCE_DATE (2025-11-24) as today
+      const today = REFERENCE_DATE.toISOString().split('T')[0];
       const stateWithDateNode: TabState = {
         tabs: [
           {
@@ -486,7 +492,8 @@ describe('TabPersistenceService', () => {
     });
 
     it('recomputes tomorrow date title correctly', () => {
-      const tomorrow = new Date();
+      // Calculate tomorrow based on REFERENCE_DATE
+      const tomorrow = new Date(REFERENCE_DATE);
       tomorrow.setDate(tomorrow.getDate() + 1);
       const tomorrowStr = tomorrow.toISOString().split('T')[0];
 
@@ -514,7 +521,8 @@ describe('TabPersistenceService', () => {
     });
 
     it('recomputes yesterday date title correctly', () => {
-      const yesterday = new Date();
+      // Calculate yesterday based on REFERENCE_DATE
+      const yesterday = new Date(REFERENCE_DATE);
       yesterday.setDate(yesterday.getDate() - 1);
       const yesterdayStr = yesterday.toISOString().split('T')[0];
 

